### DISCUSSION TITLE
How to disable copilot chat with auto-completion enabled in VSCode?

### DISCUSSION THREAD

**Discussion by gwy15** [community/community#68276] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hi there.

My VSCode copilot plugin recently auto upgraded and now copilot chat is automatically enabled. But it is ranked first in my "quick fix" menu. Now I have to either ask the slow copilot chat to get my fix hint, or press down array TWICE and select the not-so-smart but super useful and fast "Import XXXX" hint.

Please offer, or tell me how to opt-out copilot chat.

**Comment by milindaShehan** [comment, marked as answer]
To opt-out of GitHub Copilot Chat in Visual Studio Code, follow these steps:

1. Click on the **Extensions** icon in the left sidebar.
2. Search for and click on the **GitHub Copilot Chat** extension.
3. Click on the **Settings** gear icon in the bottom right corner of the extension card.
4. In the **Settings** dialog box, uncheck the **Enable** checkbox.
5. Click **OK** to save your changes.

Once you have disabled Copilot Chat, it will no longer appear in your quick fix menu.

**Alternatively, you can also disable Copilot Chat for specific languages:**

1. Click on the **Extensions** icon in the left sidebar.
2. Search for and click on the **GitHub Copilot** extension.
3. Under **Enable or disable Copilot for specified languages**, click **Edit in settings.json**.
4. In the **settings.json** file, add or remove the languages you want to enable or disable Copilot Chat for. For example, to disable Copilot Chat for Python, add the following line to the **settings.json** file:

```
"copilot.chat.enabledLanguages": ["-python"]
```

5. Save the **settings.json** file.

Once you have saved the changes, Copilot Chat will be disabled for the specified languages.

**Note:** If you are using Copilot Chat in an organization, the organization administrator may have disabled the ability to opt-out of Copilot Chat. If you are unable to disable Copilot Chat, contact your organization administrator for more information.

**Reply by gwy15** [reply]
Thank you!

**Reply by Marinell0** [reply]
As of today, this doesn't work. How can I disable the chat extension without disabling the autocomplete feature?

**Reply by barbalex** [reply]
This is very unnerving as copilot chat makes saving unbearably slow sometimes

**Reply by Namaskar-1F64F** [reply]
profiling the extensions I noticed the chat taking up a lot of time even though I never use it.

disabling / uninstalling chat disables / uninstalls copilot
at the moment the only settings are 
```
  "github.copilot.chat.welcomeMessage"
  "github.copilot.chat.useProjectTemplates"
  "github.copilot.chat.terminalChatLocation"
  "github.copilot.chat.localeOverride"
  ```
  none of which disable the chat.

**Reply by natterstefan** [reply]
As of today, I am unsure if this is what I want and need. I want to use the chat but at the same time make sure saving changes in VS Code does not take as long as it does since GitHub Copilot Chat appears in the list of code actions.

<img width="444" alt="image" src="https://github.com/community/community/assets/1043668/4dda8ce4-ae8d-4268-aa9b-e58f615e46bb">

How's it possible to reduce the time needed to save changes (again) while still being able to use the Chat (on demand)?

Thanks for your help!

PS: [Other developers](https://github.com/microsoft/vscode-eslint/issues/1055#issuecomment-2161238028) have similar issues.

**Reply by Fasteroid** [reply]
The enshittification of everything microsoft touches continues! (gasp!)

**Reply by StuartBertram** [reply]
Three months later, have we STILL not got a way to disable the chat but keep the autocomplete?

If Copilot can just give me slightly more intelligent autocomplete then it's useful. If it throws in the kitchen sink then it's not useful.

**Reply by joewright** [reply]
It auto-opened on me today and I can't figure out how to disable the chat dialogs without disabling copilot entirely. I'll likely just uninstall and unsubscribe at this point.

**Reply by Brian-at-Baebies** [reply]
It's even worse now.  I added the Outline view, and the Chat pane is part of it.  The kicker is, _I don't have the GitHub Copilot extension installed._  Not simply en****ification, but the further nonconsensual shoving of AI into my workflow.

**Reply by cherio-corro** [reply]
It marked as answered but it is not an applicable answer. I don't even have copilot extension installed and yet the chat window is ever present. I'd love not to see it.

**Reply by TelamCanidae** [reply]
SO… it 2025, I got forced feed Copilot in Code 👎 … can't find where to remove it. So I guess i will be dumping the this non sense. Open source here I come!  And who ever marked this as answered...  WOW.

**Reply by rjgotten** [reply]
@cherio-corro 
@TelamCanidae 

CTRL+SHIFT+P -> "Hide Copilot" 
Execute that command.
Confirm the dialog prompt.

Counter to the name of the command, it will actually completely disable and remove all Copilot integrations. Not hide them. Disable the whole thing. Gone. All of it. Only way to re-enable them is to opt back into trying Copilot explicitly, which the dialog prompt also informs you of.

As far as I'm aware, this is the _only_ way to excise it from VSCode in an encompassing manner since it started eating its way into every nook and cranny of the product.

**Reply by Brian-at-Baebies** [reply]
@rjgotten 😲 My hero!  Somebody give this participant a cookie.

**Reply by StuartBertram** [reply]
I wonder why it is not disabled by default and "Use AI features with Copilot" is the way to opt in to it? 🤔

**Comment by shinebayar-g** [comment]
Copilot and Copilot chat extensions install each other and can't be disabled separately? Genius design :clap: :clap:

**Reply by miclgael** [reply]
This is fun. You can't disable Copilot chat and _keep_ Copilot auto-completions. 

But if you try to hide the Copilot chat icons from the interface, VS Code crashes. 🥴

**Comment by brizaldi** [comment]
To disable the chat view from appearing every time you open VS Code, uncheck the “Show View by Default” setting.
<img width="362" height="749" alt="image" src="https://github.com/user-attachments/assets/56d0fbd1-413b-48e2-9bd8-4b35bab32d90" />
